plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/TrinoHiveCatalog.java

```diff
@@ -303,19 +304,38 @@ public List<SchemaTableName> listTables(ConnectorSession session, Optional<Strin
     @Override
     public void dropTable(ConnectorSession session, SchemaTableName schemaTableName)
     {
-        BaseTable table = (BaseTable) loadTable(session, schemaTableName);
-        TableMetadata metadata = table.operations().current();
-        validateTableCanBeDropped(table);
+        BaseTable table = null;
+        try {
+            table = (BaseTable) loadTable(session, schemaTableName);
+            validateTableCanBeDropped(table);
```

Comment:
When the metadata file is not found, Then we won't be able to get the table properties and validate if the table can be dropped. How to handle this?